tokio/src/sync/broadcast.rs

```diff
@@ -852,6 +855,10 @@ impl<T> Shared<T> {
 
             // Acquire the lock again.
             tail = self.tail.lock();
+
+            if woken >= active_receivers {
+                break;
+            }
```

Comment:
Please move this above the call to `lock` and replace the break with return. That will avoid unnecessarily locking the mutex.

Reply:
Done.